detection_rules/rule.py

```diff
@@ -1300,13 +1301,23 @@ def name(self) -> str:
         return self.contents.name
 
 
-def downgrade_contents_from_rule(rule: TOMLRule, target_version: str) -> dict:
+def downgrade_contents_from_rule(rule: TOMLRule, target_version: str, replace_id: bool = True) -> dict:
     """Generate the downgraded contents from a rule."""
-    payload = rule.contents.to_api_format()
-    meta = payload.setdefault("meta", {})
-    meta["original"] = dict(id=rule.id, **rule.contents.metadata.to_dict())
-    payload["rule_id"] = str(uuid4())
-    payload = downgrade(payload, target_version)
+    rule_dict = rule.contents.to_dict()["rule"]
```

Comment:
Yup, exactly what we discussed. `to_api_format()` would result in a dict AFTER the `transform` method was called, thus being non-compatible with the frozen `jsonschema`. 🔥